dbeam-core/src/main/java/com/spotify/dbeam/args/DbeamQueryBuilder.java

```diff
@@ -0,0 +1,184 @@
+/*-
+ * -\-\-
+ * DBeam Core
+ * --
+ * Copyright (C) 2016 - 2019 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.dbeam.args;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * Wrapper class for raw SQL query (SELECT statement).
+ */
+public class DbeamQueryBuilder implements Serializable {
+
+  private final StringBuilder sqlBuilder;
+  private final int fromIdx;
+  private Optional<String> limitStr = Optional.empty();
+
+  private DbeamQueryBuilder(final String sqlQuery) {
+    String uppSql = sqlQuery.toUpperCase();
+    if (!uppSql.startsWith("SELECT")) {
+      throw new IllegalArgumentException("Sql query should start with SELECT");
```

Comment:
This check might be too restrictive. 
Some databases permit adding settings at a query level, which influence how a query is executed.

Eg: in postgres:
```
SET work_mem = '256MB';
SELECT * FROM users ... INNER JOIN .... ;
RESET work_mem;
```

Ideally we shouldn't be parsing / validating SQL queries.

Reply:
Right. Addressed.